src/spatialdata/_io/accessor.py

```diff
@@ -0,0 +1,165 @@
+from collections.abc import Callable, Iterator, MutableMapping
+from typing import Any
+
+import dask.dataframe as dd
+from dask.dataframe.extensions import register_dataframe_accessor, register_series_accessor
+
+
+class _AttrsBase(MutableMapping[str, str | dict[str, str]]):
+    """Base accessor that stores arbitrary metadata on Dask objects."""
+
+    def __init__(self, dask_obj: dd.DataFrame | dd.Series):
+        self._obj = dask_obj
+        if not hasattr(dask_obj, "_attrs"):
+            dask_obj._attrs = {}
+
+    def __getitem__(self, key: str) -> Any:
+        return self._obj._attrs[key]
+
+    def __setitem__(self, key: str, value: str | dict[str, str]) -> None:
+        self._obj._attrs[key] = value
+
+    def __delitem__(self, key: str) -> None:
+        del self._obj._attrs[key]
+
+    def __iter__(self) -> Iterator[str]:
+        return iter(self._obj._attrs)
+
+    def __len__(self) -> int:
+        return len(self._obj._attrs)
+
+    def __repr__(self) -> str:
+        return repr(self._obj._attrs)
+
+    def __str__(self) -> str:
+        return str(self._obj._attrs)
+
+    def copy(self) -> Any:
+        return self._obj._attrs.copy()
+
+    @property
+    def data(self) -> Any:
+        """Access the raw internal attrs dict."""
+        return self._obj._attrs
+
+
+@register_dataframe_accessor("attrs")
+class DfAttrsAccessor(_AttrsBase):
+    """Dict-like .attrs accessor for Dask DataFrames."""
+
+    pass
+
+
+@register_series_accessor("attrs")
+class SeriesAttrsAccessor(_AttrsBase):
+    """Dict-like .attrs accessor for Dask Series."""
+
+    pass
+
+
+def wrap_with_attrs(method: Callable[..., Any]) -> Callable[..., Any]:
```

Comment:
I feel like these `Any` are the same no? Maybe use a generic? Same as in `def wrapper`?

Reply:
Not relevant anymore because now the code is shorter and we do need that signature.